[jit] make Future type annotation available in Python

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27637 [jit] make Future type annotation available in Python**

Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

Differential Revision: [D20626866](https://our.internmc.facebook.com/intern/diff/D20626866)

wanchaol committed:
[jit] make Future type annotation available in Python

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

wanchaol committed:
rebase on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

driazati commented on torch/jit/__init__.py:
```diff
@@ -65,7 +65,7 @@ def _parse_env(name, default, true_message, false_message):
 _python_cu = torch._C.CompilationUnit()
 
-Future = torch._C.Future
+Future = Future
```
Add a comment that this is here to satisfy `flake8` so someone doesn't delete it for looking weird

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

suo commented:
Any progress on this?

eellison commented:
Someone ran into this here, fwiw https://discuss.pytorch.org/t/torch-jit-future-type-annotation/69045

wanchaol commented:
@suo @eellison I was stuck on a python2 ci setup problem and was hoping we could remove python2 CI before I land.. Now that we still keep Python2 for a while, I will fix the CI set up to make this pass.

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

[ghstack-poisoned]

wanchaol committed:
Update on "[jit] make Future type annotation available in Python"


Summary:
Fixes https://github.com/pytorch/pytorch/issues/26578

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Pull Request resolved: https://github.com/pytorch/pytorch/pull/27637

Differential Revision: [D20626866](https://our.internmc.facebook.com/intern/diff/D20626866)

[ghstack-poisoned]
